extension-base/src/main/java/com/azure/autorest/extension/base/plugin/JavaSettings.java

```diff
@@ -66,32 +66,47 @@ public static JavaSettings getInstance()
     {
         if (_instance == null)
         {
+            String syncMethodsDefault = "essential";
+            boolean addContextParameterDefault = false;
+            boolean contextClientMethodParameterDefault = false;
+            boolean clientSideValidationsDefault = false;
+            boolean clientLoggerDefault = false;
+
+            String fluentSetting = host.getStringValue("fluent");
+            if (fluentSetting != null) {
+                syncMethodsDefault = "all";
+                addContextParameterDefault = true;
+                contextClientMethodParameterDefault = true;
+                clientSideValidationsDefault = true;
+                clientLoggerDefault = true;
+            }
+
             setHeader(host.getStringValue("license-header"));
             _instance = new JavaSettings(
-                    host.getBooleanValue("azure-arm"),
-                    host.getStringValue("fluent"),
-                    host.getBooleanValue("regenerate-pom"),
+                    host.getBooleanValue("azure-arm", false),
+                    fluentSetting,
+                    host.getBooleanValue("regenerate-pom", false),
                     _header,
                     80,
                     host.getStringValue("serviceName"),
                     host.getStringValue("namespace", "").toLowerCase(),
                     host.getBooleanValue("enable-xml", false),
                     host.getBooleanValue("non-null-annotations", false),
-                    host.getBooleanValue("client-side-validations", false),
+                    host.getBooleanValue("client-side-validations", clientSideValidationsDefault),
                     host.getStringValue("client-type-prefix"),
-                    host.getBooleanValue("generate-client-interfaces", true),
-                    host.getBooleanValue("generate-client-as-impl", true),
```

Comment:
Previously the `defaultValue` for `getBooleanValue` will never work. The underlying `getBooleanValue` will never return `null`.